libs/vectorise/include/vectorise/uint/uint.hpp

```diff
@@ -1107,23 +1125,39 @@ inline double ToDouble(UInt<256> const &x)
   return conv.value;
 }
 
-template <typename T, uint16_t S>
-constexpr void Serialize(T &s, UInt<S> const &u)
+}  // namespace vectorise
+
+namespace serializers {
+template <typename D, uint16_t S>
+struct ArraySerializer<vectorise::UInt<S>, D>
 {
-  for (std::size_t i = 0; i < u.elements(); i++)
+public:
+  using Type       = vectorise::UInt<S>;
+  using DriverType = D;
+
+  template <typename Constructor>
+  static void Serialize(Constructor &array_constructor, Type const &u)
   {
-    s << u.ElementAt(i);
+    auto array = array_constructor(u.elements());
```

Comment:
I think I would add the size of the uint in the first element here in a similar way to the bitvector

Reply:
You mean that you'd like to store `sizeof(WideType)` as first element?